### ISSUE TITLE
Iterator problem in TorchText 0.3

### ISSUE THREAD

**Issue by LividWo** [pytorch/text#328]
I have a project which worked fine with the previous version of Pytorch and TorchText. Now I migrate to the latest version, some problem with Iterator occurs.

# Code:
```
TEXT = data.Field(lower=True)
LABEL = data.Field()

train, dev, test = data.TabularDataset.splits(
    path=args.data_dir, train='train.txt', validation='valid.txt',
    test='test.txt', format='tsv',
    fields=[('id', None), ('sub', None), ('sub name', None), ('relation', None),
            ('obj', None), ('text', TEXT), ('label', LABEL)]
)
TEXT.build_vocab(train, dev, test)
LABEL.build_vocab(train, dev, test)
print(train[2].text)
print(train[3].label)
```
# This part still looks fine with the output:
```
['what', 'country', 'was', 'the', 'film', 'the', 'debt', 'from']
['O', 'O', 'O', 'I', 'I', 'O', 'O']
```

# However, applied Iterator on it : 
`train_iter = data.Iterator(train, batch_size=args.batch_size, device=args.gpu, train=True, repeat=False,
                                   sort=False, shuffle=True)
`
# I got some wrong paddings like the one below. The sentence after padding is a mess:

`what whats which what what yoo what what what who which what which bandai what which who who what what where to who what what khiladi what what where what name what
`
